src/components/flow/routers/webhook/WebhookRouterForm.tsx

```diff
@@ -76,18 +76,41 @@ export default class WebhookRouterForm extends React.Component<
     const updates: Partial<WebhookRouterFormState> = {};
 
     let ensureEmptyHeader = false;
+    let toRemove: any[] = [];
 
     if (keys.hasOwnProperty('method')) {
       updates.method = { value: keys.method };
 
       const oldMethod = this.state.method.value.value;
       const newMethod = keys.method.value;
 
-      // if we switched from GET to non-GET or non-GET to GET, reset the body back to the default
-      if (
-        (oldMethod === Methods.GET && newMethod !== Methods.GET) ||
-        (oldMethod !== Methods.GET && newMethod === Methods.GET)
-      ) {
+      // if we don't have a content-type when switching to a PUT or POST, add one
+      if (oldMethod === Methods.GET && newMethod !== Methods.GET) {
+        const existing = this.state.headers.find(
+          (header: HeaderEntry) => header.value.name.toLowerCase() === 'content-type'
+        );
+        if (!existing) {
+          let uuid = createUUID();
+
+          // if we have an empty header, use that one
+          const lastHeader =
+            this.state.headers.length > 0
+              ? this.state.headers[this.state.headers.length - 1]
+              : null;
+          if (lastHeader && !lastHeader.value.name) {
+            uuid = lastHeader.value.uuid;
+          }
+          keys.header = { uuid, name: 'Content-Type', value: 'application/json' };
+        }
+        updates.body = { value: getDefaultBody(newMethod) };
+      } else if (oldMethod !== Methods.GET && newMethod === Methods.GET) {
+        // remove content type when resetting to a get
+        const existing = this.state.headers.find(
```

Comment:
can this live outside the if to avoid the repetition?

Reply:
it could but then it would be executed for every single change in that dialog instead of just method changes. could wrap the whole thing checking for method change which would allow that tho

Reply:
Oh I see, well fine either way.

Reply:
There was actually two bits of repetition, so i liked ur suggestion better